src/main/template/vpc.yaml

```diff
@@ -0,0 +1,442 @@
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref privateRouteTableB
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: !Ref natGatewayB
+
+  privateRouteC:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref privateRouteTableC
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: !Ref natGatewayC
+
+  privateSubnetARouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableA
+      SubnetId: !Ref privateSubnetA
+
+  privateSubnetBRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableB
+      SubnetId: !Ref privateSubnetB
+
+  privateSubnetCRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref privateRouteTableC
+      SubnetId: !Ref privateSubnetC
+
+  bastionSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: !Sub ${AWS::StackName} Bastion Security Group
+      GroupDescription: !Sub ${AWS::StackName} Bastion Security Group
+      VpcId: !Ref vpc
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: 0.0.0.0/0
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Bastion Security Group
+
+  bastionLaunchConfiguration:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      ImageId: !Ref bastionAmi
+      KeyName: !Ref keyName
+      InstanceType: !Ref bastionInstanceType
+      SecurityGroups:
+        - !Ref bastionSecurityGroup
+      UserData:
+        Fn::Base64: !Sub |
+           #!/bin/bash -xe
+           yum update -y
+           /opt/aws/bin/cfn-init -v --configsets setup --stack ${AWS::StackName} --resource bastionLaunchConfiguration --region ${AWS::Region}
+           /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource bastionAutoScalingGroup --region ${AWS::Region}
+    Metadata:
+      AWS::CloudFormation::Init:
+        configSets:
+          setup:
+            - "ssh_keys"
+            - "ntp"
+        ssh_keys:
+          files:
+            /home/ec2-user/.ssh/additional_keys:
+              source: !Ref authorizedKeys
+              owner: root
+              group: root
+              mode: 000644
+          commands:
+            register_additional_keys:
+              command: "cat /home/ec2-user/.ssh/additional_keys >> /home/ec2-user/.ssh/authorized_keys"
+        ntp:
+          packages:
+            yum:
+              ntp: []
+          services:
+            sysvinit:
+              ntpd:
+                enabled: true
+                ensureRunning: true
+                packages:
+                  yum:
+                    - ntp
+
+  bastionAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      LaunchConfigurationName: !Ref bastionLaunchConfiguration
+      MaxSize: 2
```

Comment:
any reason for `2`? I would have expected `1`.

Reply:
If the `MaxSize` is only `1`, the autoscaling group will wait for the instance to be completely terminated before spinning up a new one. Having it at `2` allows for a new instance to be started as soon as the first one is deemed dead, regardless of the termination process.